### PR TITLE
[CRM][DASH] Add the possibility of querying availability for OIDs.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
- Add support for DASH resource availability to VirtualSwitchSaiInterface.

### CODE REVIEW

**Review by kcudnik — APPROVED** [review]

**Review by kcudnik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@kcudnik could you please help to merge?

**Comment by kcudnik** [comment]
Seems like there is build error nit related to this change

File: vslib/VirtualSwitchSaiInterface.cpp [hunk 1]
```diff
@@ -847,6 +847,19 @@ sai_status_t VirtualSwitchSaiInterface::objectTypeGetAvailability(
...
     }
+    else if ((objectType == (sai_object_type_t)SAI_OBJECT_TYPE_VNET) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_ENI) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_ENI_ETHER_ADDRESS_MAP_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_INBOUND_ROUTING_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_OUTBOUND_ROUTING_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_PA_VALIDATION_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_OUTBOUND_CA_TO_PA_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_DASH_ACL_GROUP) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_DASH_ACL_RULE))
```
Comment (kcudnik) [inline vslib/VirtualSwitchSaiInterface.cpp:858]: why those specific types ? do they need to be listed explicitly, seems like all of t hem are experimental objects, maybe you could use compare ?
you could get object_info from sai_metadata_get_object_type_info(ot) (check for null) on top of the function and then use info->isexperimental flag,
or if objectType = SAI_OBJECT_TYPE_MAX

**Comment by kcudnik** [comment]
please update branch and resolve conflits

**Comment by prsunny** [comment]
@kcudnik , @oleksandrivantsiv , can we merge this?

File: vslib/VirtualSwitchSaiInterface.cpp (hunk 1 above)
Comment (oleksandrivantsiv) [reply vslib/VirtualSwitchSaiInterface.cpp:858]: those are the DASH objects that we use and want to test in CRM DASH VS tests. Not all experimental objects are resources, so info->isexperimental flag and objectType >= SAI_OBJECT_TYPE_MAX are redundant.

**Comment by oleksandrivantsiv** [comment]
@oleksandrivantsiv from my pov the PR is ready to merge.

File: vslib/VirtualSwitchSaiInterface.cpp (hunk 1 above)
Comment (kcudnik) [reply vslib/VirtualSwitchSaiInterface.cpp:858]: then use flag md->isresourcetype ?

**Review by kcudnik — APPROVED** [review]

**Pull request merged**